extensions/wikia/ArticleVideo/bower_components/html5-skin/js/components/utils.js

```diff
@@ -0,0 +1,463 @@
+    if( ! orderedItems ) { return []; }
+    var self = this;
+    var validItems = orderedItems.filter( function(item) { return self._isValid(item); } );
+    var r = this._collapse( barWidth, validItems, responsiveUIMultiple );
+    return r;
+  },
+
+  /**
+  * Find thumbnail image URL and its index that correspond to given time value
+  *
+  * @function findThumbnail
+  * @param {Object} thumbnails - metadata object containing information about thumbnails
+  * @param {Number} hoverTime - time value to find thumbnail for
+  * @param {Number} duration - duration of the video
+  * @returns {Object} object that contains URL and index of requested thumbnail
+  */
+  findThumbnail: function(thumbnails, hoverTime, duration) {
+    var timeSlices = thumbnails.data.available_time_slices;
+    var width = thumbnails.data.available_widths[0]; //choosing the lowest size
+
+    var position = Math.floor((hoverTime/duration) * timeSlices.length);
+    position = Math.min(position, timeSlices.length - 1);
+    position = Math.max(position, 0);
+
+    var selectedTimeSlice = null;
+    var selectedPosition = position;
+
+    if (timeSlices[position] >= hoverTime) {
+      selectedTimeSlice = timeSlices[0];
+      for (var i = position; i >= 0; i--) {
+        if (timeSlices[i] <= hoverTime) {
+          selectedTimeSlice = timeSlices[i];
+          selectedPosition = i;
+          break;
+        }
+      }
+    } else {
+      selectedTimeSlice = timeSlices[timeSlices.length - 1];
+      for (var i = position; i < timeSlices.length; i++) {
+        if (timeSlices[i] == hoverTime) {
+          selectedTimeSlice = timeSlices[i];
+          selectedPosition = i;
+          break;
+        } else if (timeSlices[i] > hoverTime) {
+          selectedTimeSlice = timeSlices[i - 1];
+          selectedPosition = i - 1;
+          break;
+        }
+      }
+    }
+
+    var selectedThumbnail = thumbnails.data.thumbnails[selectedTimeSlice][width].url;
+    return { url: selectedThumbnail, pos: selectedPosition };
+  },
+
+  /**
+  * Check if the current browser is on a touch enabled device.
+  * Function from https://hacks.mozilla.org/2013/04/detecting-touch-its-the-why-not-the-how/
+  *
+  * @function browserSupportsTouch
+  * @returns {Boolean} Whether or not the browser supports touch events.
+  */
+  browserSupportsTouch: function() {
+    return ('ontouchstart' in window) ||
+     (navigator.maxTouchPoints > 0) ||
+     (navigator.msMaxTouchPoints > 0);
+  },
+
+  /**
+   * Creates wrapper object with sanitized html. This marked data can subsequently be passed into dangerouslySetInnerHTML
+   * See https://facebook.github.io/react/tips/dangerously-set-inner-html.html
+   *
+   * @function createMarkup
+   * @returns {Object} Wrapper object for sanitized markup.
+   */
+  createMarkup: function(html) {
+    return {__html: html};
+  },
+
+  _isValid: function( item ) {
+    var valid = (
+      item &&
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce the number of conditional operators (5) used in the expression (maximum allowed 3). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS1067)